src/main/kotlin/org/dash/mobile/explore/sync/GCManager.kt

```diff
@@ -19,14 +19,18 @@ private const val GCS_LOCK_FILE_NAME = "explore/.lock"
 const val CHECKSUM_META_KEY = "Data-Checksum"
 const val TIMESTAMP_META_KEY = "Data-Timestamp"
 
-class GCManager {
+class GCManager(private val mode: OperationMode) {
 
     private val logger = LoggerFactory.getLogger(GCManager::class.java)!!
 
     private val gcStorage by lazy {
         createStorage()
     }
 
+    private val lockFileName by lazy {
+        "$GCS_LOCK_FILE_NAME-$mode"
+    }
+
```

Comment:
each mode will have it's own lock file, which will prevent the production build from running if the testnet or dev build is running.